### ISSUE TITLE
implement serialization

### ISSUE THREAD

**Issue by oli-obk** [serde-rs/xml#3]
deserializing xml is great, but we also need serialization.

**Comment by johansja** [comment]
Hi. This issue has been opened for more than a year. What is the progress on this? What help might be needed?

**Comment by oli-obk** [comment]
Unfortunately I have no need for serialization. Therefor I don't think I'm very qualified to make decisions on how to serialize certain things.

I think we could just copy the json `Serializer`s so we get the pretty printing for free and then adjust it to produce xml tags.

There is the big question on how to decide which things go into attributes and which go into inner tags. I don't have a good solution to this problem

**Comment by johansja** [comment]
Interesting on the last point. Can we have a custom tag for that?

Do you know anyone who might be knowledgeable in this?

**Comment by oli-obk** [comment]
we could use the `#[serde(rename="@foo")]` method that I'm using for inner text.

**Comment by johansja** [comment]
Good. I am just starting in Rust and maybe this will be a good starting point for me too. What do you think?

**Comment by oli-obk** [comment]
I'm happy for any contributions. If you have any questions, just ask. I'm also available on irc.mozilla.org on #serde

**Comment by softprops** [comment]
No progress on this yet?

**Comment by oli-obk** [comment]
Nope, gonna do some serde 0.9 changes first + probably break a lot of code, because the current design is somewhat unfortunate.

**Comment by blanham** [comment]
@oli-obk What's the current status of this?

**Comment by oli-obk** [comment]
This entire repository is deprecated in favour of serde-xml-rs which does serialization

**Comment by RReverser** [comment]
@oli-obk I'm afraid that's not correct, serde-xml-rs currently implements only deserialization. There was some unfinished attempt in a PR, but that's it for now.
